src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -6345,28 +6386,69 @@ public void CreateTupleTypeSymbol_ElementTypes_DefaultArgs()
-            Assert.True(tuple1.Equals(tuple2));
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
 
             tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementNames: default);
-            Assert.True(tuple1.Equals(tuple2));
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
 
             tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementLocations: default);
-            Assert.True(tuple1.Equals(tuple2));
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
 
             tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementNullableAnnotations: default);
-            Assert.True(tuple1.Equals(tuple2));
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+        }
+
+        [Fact]
+        [WorkItem(36047, "https://github.com/dotnet/roslyn/issues/36047")]
+        [WorkItem(40105, "https://github.com/dotnet/roslyn/issues/40105")]
+        public void CreateTupleTypeSymbol_ElementTypes_DefaultArgs_02()
+        {
+            var source =
+@"class Program
+{
+    (string,
+#nullable enable
+             string, string?) F;
+}";
+            var comp = (Compilation)CreateCompilation(source);
+            var tuple1 = (INamedTypeSymbol)((IFieldSymbol)comp.GetMember("Program.F")).Type;
+            var elementTypes = tuple1.TupleElements.SelectAsArray(e => e.Type);
+
+            var tuple2 = comp.CreateTupleTypeSymbol(elementTypes);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, default, default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, default, default, default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, default(ImmutableArray<string>), default(ImmutableArray<Location>), default(ImmutableArray<CodeAnalysis.NullableAnnotation>));
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementNames: default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementLocations: default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementNullableAnnotations: default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
         }
```

Comment:
>} [](start = 8, length = 1)

Are we testing cases where `elementTypes` has annotations and so does `elementNullableAnnotations`?

Reply:
> Are we testing cases where elementTypes has annotations and so does elementNullableAnnotations?

I believe existing tests below sufficiently cover these cases.

---
In reply to: [357434927](https://github.com/dotnet/roslyn/pull/40348#discussion_r357434927) [](ancestors = 357434927)

Reply:
I couldn't find any test where the annotations on `elementTypes` are not `None` and distinct from `elementNullableAnnotations`.

---
In reply to: [357436843](https://github.com/dotnet/roslyn/pull/40348#discussion_r357436843) [](ancestors = 357436843,357434927)

Reply:
> I couldn't find any test where the annotations on elementTypes are not None and distinct from elementNullableAnnotations.

This PR is focused on scenarios where ```elementNullableAnnotations``` is default.

---
In reply to: [357776096](https://github.com/dotnet/roslyn/pull/40348#discussion_r357776096) [](ancestors = 357776096,357436843,357434927)